### PR TITLE
Document string support

### PR DESCRIPTION
**Pull request by seibert**
Documentation for preliminary string support

### CODE REVIEW

File: docs/source/reference/pysupported.rst [hunk 1]
```diff
@@ -144,6 +144,44 @@ The following attributes and methods are supported:
 * ``.real``
 * ``.imag``
 
+str
+---
+
+Numba supports (Unicode) strings in Python 3.  Strings can be passed into
+nopython mode as arguments, as well as constructed and returned from nopython
```
Comment (stuartarchibald) [inline docs/source/reference/pysupported.rst]:
```suggestion
:term:`nopython mode` as arguments, as well as constructed and returned from :term:`nopython mode`
```

File: docs/source/reference/pysupported.rst [hunk 2]
```diff
@@ -144,6 +144,44 @@ The following attributes and methods are supported:
...
+mode.  As in Python, slices (even of length 1) return a new, reference counted
```
Comment (stuartarchibald) [inline docs/source/reference/pysupported.rst]:
```suggestion
.  As in Python, slices (even of length 1) return a new, reference counted
```

**Comment by seibert** [comment]
OK, I have accepted the proposed change.

**Comment by stuartarchibald** [comment]
hmmm, that edit functionality on github doesn't line wrap or deal with white space, guess that's expected, so apologies, this needs manual fixing now. The change renders ok as HTML (apart from a space before the full stop), but the source lines overrun. Perhaps edit manually?! Guess we've learned not to do that again for RST.

**Pull request merged**